README.md

```diff
@@ -1 +1,55 @@
-# phase_zinc
\ No newline at end of file
+# phase_zinc
+
+## Contents
+*  [Overview](#overview)
+*  [Files included](#files-included)
+*  [How to set up](#how-to-set-up)
+
+## Overview
+A custom phase extension allowing [bazelbuild/rules_scala](https://github.com/bazelbuild/rules_scala) users to use the [Zinc compiler](https://github.com/sbt/zinc). To learn what "customizable phases" are and how the phase architecture can help your Bazel project, see [this doc on customizable phases](https://github.com/bazelbuild/rules_scala/blob/master/docs/customizable_phase.md) on the [bazelbuild/rules_scala](https://github.com/bazelbuild/rules_scala) project.
+
+This project is aimed at anyone who depends on the [bazelbuild/rules_scala](https://github.com/bazelbuild/rules_scala) project but would like to swap out the default Scala compiler with Zinc compiler. This is especially helpful for codebases that have Bazel targets with large numbers of source files-- if Bazel targets are not granular enough, builds can take a long time. Because Zinc is an incremental compiler, it can make builds significantly faster. However, be aware that Zinc can also add non-determinism to your builds.
+
+## Files included
+
+Phase_zinc defines a Zinc compiler phase, then adds this phase to three Bazel rules originally defined in [bazelbuild/rules_scala](https://github.com/bazelbuild/rules_scala): `scala_binary`, `scala_library` and `scala_test`. The new corresponding rules are named, respectively, `zinc_scala_binary`, `zinc_scala_library` and `zinc_scala_test`.
+
+The [rules](rules) directory contains definitions and implementations of the Zinc compiler phase. [rules/phase/phase_zinc_compile.bzl](rules/phase/phase_zinc_compile.bzl) defines the logic of how the phase works. [rules/ext/phase_zinc_compile_ext.bzl](rules/ext/phase_zinc_compile_ext.bzl) wraps the phase definition in an extension together with some extra attributes. Finally, [rules/scala.bzl](rules/scala.bzl) passes the extension to a rule macro (e.g. `make_scala_binary`) to add the phase to a rule (e.g. creating `zinc_scala_binary`).
+ 
+The [test](test) directory contains Scala source code to test functionality of the phase.
+
+The [src/main/scala](src/main/scala) directory contains Scala and Zinc configuration data. See [src/main/scala/workers/BUILD.bazel](src/main/scala/workers/BUILD.bazel).
+
+The [scripts](scripts) directory contains shell scripts to run these tests. Run `scripts/travis.sh` to execute all tests.
```

Comment:
I don't think it's necessary. Users shouldn't care about how to run tests nor how we set up the CI. Plus, if we want to merge it back to `bazelbuild/rules_scala`, we will discard all scripts. We should use their scripts.

Reply:
As far as I know, our current goal is not to merge back into `bazelbuild/rules_scala`. We just want to add an external link from `bazelbuild/rules_scala` to this repository.

Reply:
I also thought that having explanations for the different directories will help first-time users know what is going on.